File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.cpp
class Solution {
public:
    int findLucky(vector<int>& arr) {
        int ans = -1;
        unordered_map<int, int> mp;

        for(auto it : arr) {
            mp[it]++;
        }

        for(auto it : mp) {
            if(it.first == it.second) ans = max(ans, it.first);
        }

        return ans;
    }
};

File: 1510-find-lucky-integer-in-an-array/1510-find-lucky-integer-in-an-array.java
class Solution {
    public int findLucky(int[] arr) {
        int ans = -1;
        HashMap<Integer, Integer> mp = new HashMap<>();
        for(int it : arr) {
            mp.put(it, mp.getOrDefault(it, 0) + 1);
        }

        for(int key : mp.keySet()) {
            if(key == mp.get(key)) {
                ans = Math.max(ans, key);
            }
        }

        return ans;
    }
}